src/utils/__tests__/access_log.test.ts

```diff
@@ -0,0 +1,225 @@
+import {
+  aSAMLRequest,
+  aSAMLResponse,
+  aSAMLResponseWithoutRequestId,
+  aSAMLResponseWithoutFiscalCode,
+} from "../../__mocks__/spid";
+import { DOMParser } from "xmldom";
+import {
+  getFiscalNumberFromPayload,
+  getRequestIDFromPayload,
+  getRequestIDFromRequest,
+  getRequestIDFromResponse,
+  storeSpidLogs,
+} from "../access_log";
+import { BlobService } from "azure-storage";
+import {
+  FiscalCode,
+  IPString,
+  NonEmptyString,
+} from "@pagopa/ts-commons/lib/strings";
+import { SpidLogMsg } from "../../types/access_log";
+import { left, right, toError } from "fp-ts/lib/Either";
+import { fromLeft, taskEither } from "fp-ts/lib/TaskEither";
+import { Option, some } from "fp-ts/lib/Option";
+
+// Mock encrypt module to spy on toEncryptedPayload function
+jest.mock("@pagopa/ts-commons/lib/encrypt", () => ({
+  __esModule: true,
+  ...jest.requireActual("@pagopa/ts-commons/lib/encrypt"),
+  toEncryptedPayload: jest.fn((_, __) =>
+    right(({
+      cypherText: "encoded",
+      iv: "iv",
+      encryptedKey: "ek",
+    } as unknown) as encrypt.EncryptedPayload)
+  ),
+}));
+import * as encrypt from "@pagopa/ts-commons/lib/encrypt";
+const spiedToEncryptedPayload = jest.spyOn(encrypt, "toEncryptedPayload");
```

Comment:
why not spy directly without jest.mock ? i.e.
```
const toEncryptedPayloadMock = jest.fn()
.mockImplementation((_,__) => right(({
      cypherText: "encoded",
      iv: "iv",
      encryptedKey: "ek",
    } as unknown) as encrypt.EncryptedPayload)
  );
jest.spyOn(encrypt, "toEncryptedPayload")
    .mockImplementation(toEncryptedPayloadMock);
```

Reply:
With this module it's not going to work. It's like we cannot mock just the spied function and it's not returning an either.